lib/platform/linux-i2c.c

```diff
@@ -48,10 +49,44 @@ struct switchtec_i2c {
 	int i2c_addr;
 };
 
+#define CMD_GET_CAP  0xE0
+#define CMD_GET_WRITE_STATUS  0xE2
+#define CMD_GAS_WRITE_WITH_STATUS  0xE8
+#define CMD_GAS_READ  0xE9
+#define CMD_GAS_WRITE  0xEA
+
+#define MAX_RETRY_COUNT  100
+
+#define PEC_BYTE_COUNT  1
+#define DATA_TAIL_BYTE_COUNT  2
+
+#define CAP_MASK  0x80
+#define GAS_TWI_MRPC_ERR  0x20
+
 #define to_switchtec_i2c(d)  \
 	((struct switchtec_i2c *) \
 	 ((char *)d - offsetof(struct switchtec_i2c, dev)))
 
+static uint8_t gas_write_tag = 0;
+
```

Comment:
This is not a good idea. What happens if code using this library opens two instances of the I2C device? They will both share the tag and weird things might happen.  You should put it in 'struct switchtec_i2c' instead.